synapse/models/inet.py

```diff
@@ -408,12 +410,51 @@ def _normPyStr(self, valu):
         except Exception as e:
             raise s_exc.BadTypeValu(valu=valu, name=self.name, mesg=str(e))
 
+class IPv4Range(s_types.Range):
+    _opt_defs = ()
+
+    def postTypeInit(self):
+        self.opts['type'] = ('inet:ipv4', {})
+        s_types.Range.postTypeInit(self)
+        self.setNormFunc(str, self._normPyStr)
+        self.cidrtype = self.modl.type('inet:cidr4')
+
+    def _normPyStr(self, valu):
+        if '-' in valu:
+            return super()._normPyStr(valu)
+        cidrnorm = self.cidrtype._normPyStr(valu)
+        tupl = cidrnorm[1]['subs']['network'], cidrnorm[1]['subs']['broadcast']
+        return self._normPyTuple(tupl)
+
+    def _normPyTuple(self, valu):
```

Comment:
This is just a copy from the base Range class. Is there any reason to not just inherit that one and remove this?

Reply:
good call